digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -1553,7 +1553,11 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
             setProjectStatus(projectId, Sent2TR, withSession = false)
             logger.info(s"Sending to TR successful: ${trProjectStateMessage.reason}")
             PublishResult(validationSuccess = true, sendSuccess = true, Some(trProjectStateMessage.reason))
-
+          case 500 =>
+            logger.error(s"Sending project ${trProjectStateMessage.projectId} to TR failed. Status: ${trProjectStateMessage.status} Error message: ${trProjectStateMessage.reason}")
+            val returningMessage = if (trProjectStateMessage.reason.nonEmpty) trProjectStateMessage.reason else TrConnectionError
+            projectDAO.updateProjectStatus(projectId, SendingToTR)
```

Comment:
the main difference seems to be the update in case there is error 500. So in case we got no connection to TR the trProjectStateMessage.reason will be empty. Could we then change the **case _** to 
val returningMessage = if (trProjectStateMessage.reason.nonEmpty) trProjectStateMessage.reason else {
projectDAO.updateProjectStatus(projectId, SendingToTR)
TrConnectionError
} ?

Reply:
I think there is a difference: there can be connection available to TR, but not available TR execution, then TR returns that 500 in response object. And that response. might have some additional information ?
If there is no connection available ,no response object is returned , then catch() handles that situation ?